src/main/drivers/flash_w25n02k.c

```diff
@@ -0,0 +1,969 @@
+}
+
+/****************** 触发Flash设备上的页面编程操作 ******************/
+static void w25n02kv_programExecute(flashDevice_t *fdevice, uint32_t pageAddress)
+{
+    // 确保W25N02KV设备就绪
+    w25n02kv_waitForReady(fdevice);
+
+    // 执行程序操作指令，传递页地址
+    w25n02kv_performCommandWithPageAddress(&fdevice->io, W25N02KV_INSTRUCTION_PROGRAM_EXECUTE, pageAddress);
+
+    // 设置超时，保证程序执行完成
+    w25n02kv_setTimeout(fdevice, W25N02KV_TIMEOUT_PAGE_PROGRAM_MS);
+}
+
+
+//
+// Writes are done in three steps:
+// (1) Load internal data buffer with data to write
+//     - We use "Random Load Program Data", as "Load Program Data" resets unused data bytes in the buffer to 0xff.
+//     - Each "Random Load Program Data" instruction must be accompanied by at least a single data.
+//     - Each "Random Load Program Data" instruction terminates at the rising of CS.
+// (2) Enable write
+// (3) Issue "Execute Program"
+//
+
+/*
+flashfs page program behavior
+- Single program never crosses page boundary.
+- Except for this characteristic, it program arbitral size.
+- Write address is, naturally, not a page boundary.
+
+To cope with this behavior.
+
+pageProgramBegin:
+If buffer is dirty and programLoadAddress != address, then the last page is a partial write;
+issue PAGE_PROGRAM_EXECUTE to flash buffer contents, clear dirty and record the address as programLoadAddress and programStartAddress.
+Else do nothing.
+
+pageProgramContinue:
+Mark buffer as dirty.
+If programLoadAddress is on page boundary, then issue PROGRAM_LOAD_DATA, else issue RANDOM_PROGRAM_LOAD_DATA.
+Update programLoadAddress.
+Optionally observe the programLoadAddress, and if it's on page boundary, issue PAGE_PROGRAM_EXECUTE.
+
+pageProgramFinish:
+Observe programLoadAddress. If it's on page boundary, issue PAGE_PROGRAM_EXECUTE and clear dirty, else just return.
+If pageProgramContinue observes the page boundary, then do nothing(?).
+*/
+
+static uint32_t programStartAddress;
+static uint32_t programLoadAddress;
+static bool bufferDirty = false;
+static bool isProgramming = false;
+
+/****************** 开始Flash设备上的页面编程过程 ******************/
+void w25n02kv_pageProgramBegin(flashDevice_t *fdevice, uint32_t address, void (*callback)(uint32_t length))
+{
+    fdevice->callback = callback;  // 保存回调函数，用于操作完成后调用
+
+    if (bufferDirty) {
+        if (address != programLoadAddress) {
+            w25n02kv_waitForReady(fdevice);  // 等待设备就绪
+
+            isProgramming = false;
+
+            w25n02kv_writeEnable(fdevice);  // 启用写操作
+
+            w25n02kv_programExecute(fdevice, W25N02KV_LINEAR_TO_PAGE(programStartAddress));  // 执行页面编程
+
+            bufferDirty = false;
+            isProgramming = true;
+        }
+    } else {
+        programStartAddress = programLoadAddress = address;  // 设置开始和加载地址
+    }
+}
+
+/****************** 继续W25N02KV设备上的页面编程过程 ******************/
+uint32_t w25n02kv_pageProgramContinue(flashDevice_t *fdevice, uint8_t const **buffers, uint32_t *bufferSizes, uint32_t bufferCount)
```

Comment:
This will suffer from very poor blocking performance. This needs changes made equivalent to https://github.com/betaflight/betaflight/pull/13555 and https://github.com/betaflight/betaflight/pull/13562. I’ll take a closer look later.